tests/src/testfebases.cpp

```diff
@@ -0,0 +1,108 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include <config.h>
+
+#include "testhelpers.hh"
+
+#include <dune/common/test/testsuite.hh>
+#include <dune/functions/functionspacebases/basistags.hh>
+#include <dune/functions/functionspacebases/compositebasis.hh>
+#include <dune/functions/functionspacebases/lagrangebasis.hh>
+#include <dune/functions/functionspacebases/powerbasis.hh>
+#include <dune/grid/yaspgrid.hh>
+
+#include <ikarus/finiteelements/febases.hh>
+
+using Dune::TestSuite;
+
+#include <ikarus/utils/basis.hh>
+#include <ikarus/utils/init.hh>
+
+template <typename GridView, typename PreBasis>
+auto getBasis(const GridView& gridView, const PreBasis& preBasis) {
+  const auto basis      = Ikarus::makeBasis(gridView, preBasis);
+  const auto& localView = basis.flat().localView();
+  auto element          = elements(gridView).begin();
+  return Ikarus::FEBases<decltype(basis)>(basis, *element);
+}
+
+template <typename TestSuiteType, typename... Bases>
+void checkBases(TestSuiteType& t, const std::vector<int>& expectedNumberOfChildren,
+                const std::vector<int>& expectedSize, const std::string& basisName, const Bases&... bases) {
+  assert(sizeof...(bases) == expectedNumberOfChildren.size() && "Input size mismatch for number of children.");
+  assert(sizeof...(bases) == expectedSize.size() && "Input size mismatch for local view size.");
+  int i = 0;
+  (
+      [&] {
+        checkScalars(t, bases.numberOfChildren(), expectedNumberOfChildren[i],
+                     " Number of children is incorrect for i = " + std::to_string(i) + basisName);
+        checkScalars(t, static_cast<int>(bases.size()), expectedSize[i],
+                     " Size is incorrect for i = " + std::to_string(i) + basisName);
+        std::vector<typename std::remove_cvref_t<decltype(bases.localView())>::MultiIndex> dofs;
+        bases.globalFlatIndices(dofs);
```

Comment:
You can check a lot more here. 
First you can sort the indices with `std::ranges::sort` and then check for duplicates with ```cpp
bool hasDuplicates = std::adjacent_find(sortedVector.begin(), sortedVector.end()) != sortedVector.end();
```